File: php/validate_list.php
<?php
  require_once 'dv3_api.php';
  define('API_KEY', $argv[1]);

  $list_name = $argv[2];
  $file_name = $argv[3];
  assert(is_file($file_name));

  echo("Getting upload URL...\n");
  $upload_url = get_upload_url($list_name, 0, false);
  echo("Upload URL: $upload_url\n");
  echo("Uploading file contents\n");
  $list_id = upload($upload_url, $file_name);
  echo("List created. List id: $list_id\n");

  while (true) {
    $info = get_list_info($list_id);
    if ($info->{'status_value'} == 'VALIDATED') {
      echo("List validated. Grade summary: " . var_export($info->{'grade_summary'}, true) . "\n");
      break;
    } else if ($info->{'status_value'} == 'FAILED') {
      echo("List validation failed!\n");
      break;
    }

    echo("Status: {$info->{'status_value'}} ({$info->{'status_percent_complete'}} %)\n");
    sleep(5);
  }

File: python/validate_list.py
#!/usr/bin/env python3
# coding: utf-8
from __future__ import absolute_import, division, print_function
import logging
import os
import sys
import time

import dv3_api

logging.basicConfig(level=logging.INFO)
dv3_api.KEY = sys.argv[1]

list_name, file_path = sys.argv[2:4]  # pylint: disable=unbalanced-tuple-unpacking
assert os.path.isfile(file_path)

logging.info('Getting upload URL...')
upload_url = dv3_api.get_upload_url(list_name, email_column=0, has_header=False)
logging.info('Upload URL: %s', upload_url)
logging.info('Uploading file contents')
list_id = dv3_api.upload(upload_url, file_path)
logging.info('List created. Id: %s', list_id)

while True:
    time.sleep(5)
    info = dv3_api.get_list_info(list_id)
    if info['status_value'] == 'VALIDATED':
        logging.info('List validated. Grade summary: %s', info['grade_summary'])
        break
    elif info['status_value'] == 'FAILED':
        logging.info('List validation failed!')
        break

    logging.info('Status: %s (%.2f%%)', info['status_value'], info['status_percent_complete'])
